### DISCUSSION TITLE
Requesting better docs on the bench command

### DISCUSSION THREAD

**Discussion by jamesbraza** [official-stockfish/Stockfish#4424] [Q&A] [answered]
Looking at the Wiki for `bench`: https://github.com/official-stockfish/Stockfish/wiki/Commands#bench

What is the benchmark actually marking?  Is it just reporting how well one's computer does when running Stockfish?

---

Why am I asking?

I was wondering if I could use the `bench` command to analyze a chess player's skill level.  However, from reading the source [`.cpp` file](https://github.com/official-stockfish/Stockfish/blob/sf_15.1/src/benchmark.cpp#L105) it seems that `bench` is more of a computation benchmark for how well one can run the engine.

**Comment by Disservin** [comment, marked as answer]
It just got updated by @dubslow , thanks btw  ^^

**Comment by dubslow** [comment]
Indeed, how does it look now @jamesbraza ?

**Reply by jamesbraza** [reply]
Yeah looks great, definitely answers my question.  Also, imo the `bench` command is indicative of the level of competency/insight the Stockfish dev's have, I didn't realize.  Cheers!

**Reply by dubslow** [reply]
> imo the bench command is indicative of the level of competency/insight the Stockfish dev's have

what on earth is this supposed to mean? it's just random, it's almost like a one-way hash function.

**Reply by jamesbraza** [reply]
Lol @dav1312  why am I getting a thumbs down for a compliment?

I just thought it's smart to index an algorithm based on it's performance, instead of the implementation.  Anyways, back to coding...

**Reply by dubslow** [reply]
I share dav's reaction. It's a confusing comment that can easily be interpreted as "the nodecount signature of a commit is a direct reflection of the author's (in)competence", which is a grossly false notion.

I assume that's not what you mean tho.

>  to index an algorithm based on it's performance, instead of the implementation

I'm still not sure what this is supposed to mean. We use the nodecount signature to track which implementation is which. Speed ("performance") has nothing to do with it.

This usage of nodecount signature to verify implementation version is very old and widespread across the chess engine world. It is by no means unique to Stockfish.

**Reply by dubslow** [reply]
All that said, I'm glad to hear that the wiki edits fully addressed the initial issue, thanks for the feedback.

**Reply by jamesbraza** [reply]
Too bad on GitHub it's easy to misinterpret messages/tone, I genuinely was trying to speak highly of you all.  Sorry it was worded such that it came off as sarcastic.

Thanks again for the quick turnaround, case closed!
